nes-data-types/include/DataTypes/Schema.hpp

```diff
@@ -128,6 +128,27 @@ class Schema
 /// Returns a copy of the input schema without any source qualifier on the schema fields
 Schema withoutSourceQualifier(const Schema& input);
 
+/// Checks if two schemas are equal and keeps track of every mismatch and missing / unexpected fields
+struct SchemaDiff
+{
+    struct SchemaMismatch
+    {
+        size_t index{};
+        Schema::Field expectedField;
+        Schema::Field actualField;
+    };
```

Comment:
Given that the schema diff is order-sensitive, we can make the check a lot easier by simply asserting that the size matches and that each field matches. 

It is probably overkill to try to guess what order could be the correct order and thus determine which fields are missing or additional